python/pyspark/ml/functions.py

```diff
@@ -522,7 +529,11 @@ def predict_columnar(x1: np.ndarray, x2: np.ndarray) -> Mapping[str, np.ndarray]
         output), a dictionary of named numpy arrays (for multiple outputs), or a row-oriented list
         of dictionaries (for multiple outputs).
     return_type : :class:`pspark.sql.types.DataType` or str.
-        Spark SQL datatype for the expected output.
+        Spark SQL datatype for the expected output:
+        - ArrayType --> 2-dim numpy array.
+        - StructType --> dict with keys matching struct fields.
+        - StructType --> list of dict with keys matching struct fields, for models like the
```

Comment:
One more comment: pls also add description for the scalar type ( IntegerType / DoubleType etc.) case.